lib/jobs/get-catalog-values-job.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+var di = require('di');
+
+module.exports = getCatalogValuesJobFactory;
+di.annotate(getCatalogValuesJobFactory, new di.Provide('Job.Get.Catalog.Values'));
+di.annotate(getCatalogValuesJobFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Services.Waterline',
+    'Util',
+    '_'
+));
+
+function getCatalogValuesJobFactory(
+    BaseJob,
+    Logger,
+    waterline,
+    util,
+    _
+){
+    var logger = Logger.initialize(getCatalogValuesJobFactory);
+
+    /**
+     * This job finds one or more requested values from one or more specified catalogs.
+     * The catalog values are made available in the shared context in an object that 
+     * maps the catalog values to requester provided keys.
+     *
+     * Example requestedData array:
+     * [{
+     *   "source": "ohai",
+     *   "keys": {
+     *     "myIPAddrKeyName": "data.ipaddress"
+     *   }
+     * }]
+     *
+     * Adds to context.data:
+     * {
+     *   "myIPAddrKeyName": <catalog value>
+     * }
+     *
+     * @param options
+     * @param context
+     * @param taskId
+     * @constructor
+     */
+
+    function GetCatalogValuesJob(options, context, taskId){
+        GetCatalogValuesJob.super_.call(this, logger, options, context, taskId);
+        this.nodeId = context.target;
+        this.options = options;
+        this.logger = logger;
+    }
+
+    util.inherits(GetCatalogValuesJob, BaseJob);
+
+    /**
+     * @memberOf GetCatalogValuesJob
+     */
+    GetCatalogValuesJob.prototype._run = function _run(){
+        var self = this,
+            newObj = {};
+
+        // Get the correct catalog based on source
+        _.forEach(self.options.requestedData, function(catalogInfo){
```

Comment:
That's interesting because when I run the graph it is actually going through each item in the array just fine. However, it sounds like that is not the desired behavior, so regardless I need to implement it differently. I'll change the `_.forEach` to `Promise.each` so that I'm only calling `_done` once. It is definitely a better solution. Thank you for the feedback.

If you are interested... I was curious so I did some investigating. It seems the reason mine behaves as I expected, excluding the  multiple calls to `_done`, is related  to how the `_.mapValues`  & `_.assign` behave. The  `_.assign` to update the newObj isn't called until all the values are mapped to the keys. 

I added some logging into the the Job:
```
 GetCatalogValuesJob.prototype._run = function _run(){
        var self = this,
            newObj = {};

        // Get the correct catalog based on source
        _.forEach(self.options.requestedData, function(catalogInfo){
            waterline.catalogs.findMostRecent({
                node: self.nodeId, source: catalogInfo.source
            })
                .then(function(catalog){
                    _.assign(newObj, _.mapValues(catalogInfo.keys, function(val){
                        console.log('*** What is my val ' + val);
                        return _.get(catalog, val, null);
                    }));
                })
                .then(function(){
                    self.context.data = newObj;
                    console.log('*** CATALOG *** ' + catalogInfo.source + '\n***** CONTEXT.DATA ***** ' + JSON.stringify(self.context.data));
                    self._done();
                })
                .catch(function(err){
                    self._done(err);
                });
        });
    };
```
And when I run the consuming task it produces the following:
```
job:    Job.Get.Catalog.Values
*** What is my val data.cpu.0.cores
*** What is my val data.ipaddress
*** CATALOG *** ohai
***** CONTEXT.DATA ***** {"cpu_cores":"1","ip":"172.31.128.2"}
*** What is my val data.handle
*** CATALOG *** lshw
***** CONTEXT.DATA ***** {"cpu_cores":"1","ip":"172.31.128.2","test_val":"DMI:0001"}
*** What is my val [data][Chassis Information][Type]
*** CATALOG *** dmi
***** CONTEXT.DATA ***** {"cpu_cores":"1","ip":"172.31.128.2","test_val":"DMI:0001","two_words":"Other"}
[info] [2016-09-28T16:54:08.799Z] [on-taskgraph] [TaskGraph.TaskRunner] [Server] Task finished
```